src/Paprika/Data/SlottedArray.cs

```diff
@@ -62,7 +62,8 @@ public bool TrySet(in NibblePath key, ReadOnlySpan<byte> data)
 
     private bool TrySetImpl(ushort hash, byte preamble, in NibblePath trimmed, ReadOnlySpan<byte> data)
     {
-        if (TryGetImpl(trimmed, hash, preamble, out var existingData, out var index))
+        var index = TryGetImpl(trimmed, hash, preamble, out var existingData);
```

Comment:
maybe better that it would return a `bool` and `index` would be out param?

Reply:
This was done before. As https://github.com/NethermindEth/Paprika/pull/369 showed it cuts off 40 bytes from 800 byte long method without impact on the call sites that need to compare it anyway.